src/lib/nested_property.js

```diff
@@ -118,19 +120,38 @@ function npGet(cont, parts) {
  * The ONLY case we are looking for is where the entire array is selected, parts[end] === 'x'
  * AND the replacement value is an array.
  */
-function isDataArray(val, key) {
+// function isNotAContainer(key) {
+//     var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'];
 
-    var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'],
-        isNotAContainer = containers.indexOf(key) === -1;
+//     return containers.indexOf(key) === -1;
+// }
 
-    return isArray(val) && isNotAContainer;
+/*
+ * Can this value be deleted? We can delete any empty object (null, undefined, [], {})
+ * EXCEPT empty data arrays. If it's not a data array, it's a container array,
+ * ie containing objects like annotations, buttons, etc
+ */
+var DOMAIN_RANGE = /(^|.)(domain|range)$/;
+function isDeletable(val, propStr) {
```

Comment:
Alright, letting every array be deletable and running the full test suite was fairly enlightening - There are a lot of things that fail or error out, some of which could be addressed but fundamentally for data arrays there really is a difference between `[]` (which means there is data of that kind, it's just presently empty) and `undefined` (which means the data/attribute should get some default value filled in).

As far as I can tell, for `info_array` attributes there is no difference between `[]` and `undefined` though. So unless we want to try and build in here some way to hunt through the schema (which sounds super painful to me, and potentially not even well-defined given that `nestedProperty` can, on occasion, be called on a root object that's neither a complete trace or layout object but some portion of it) I would argue for some simple tests that might frequently describe an info array as falsely *not* deletable but will never describe a real data array as deletable.